In [2]:
# Dependencies
import pandas as pd
import plotly.express as px
import os
from scipy.stats import linregress
from sklearn import datasets
import statsmodels.api as sm
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
bikecounts_file = os.path.join('../','Resources', 'bikecounts.csv')
bikecounts_df = pd.read_csv(bikecounts_file)
bikecounts_df['Day'] = pd.to_datetime(bikecounts_df['Day'])
bikecounts_df.head()

Unnamed: 0        Date        Day  High Temp (°F)  Low Temp (°F)  \
0           0  2016-04-01 2016-04-01            78.1           66.0   
1           1  2016-04-02 2016-04-02            55.0           48.9   
2           2  2016-04-03 2016-04-03            39.9           34.0   
3           3  2016-04-04 2016-04-04            44.1           33.1   
4           4  2016-04-05 2016-04-05            42.1           26.1   

   Precipitation  Brooklyn Bridge  Manhattan Bridge  Williamsburg Bridge  \
0           0.01           1704.0              3126               4115.0   
1           0.15            827.0              1646               2565.0   
2           0.09            526.0              1232               1695.0   
3           0.47            521.0              1067               1440.0   
4           0.00           1416.0              2617               3081.0   

   Queensboro Bridge  Total  
0             2552.0  11497  
1             1884.0   6922  
2             1306.0   4759  
3             1307.0   4335  
4             2357.0   9471

In [4]:
bikecounts_df = pd.DataFrame(bikecounts_df)

In [5]:
bikecounts_df = bikecounts_df.rename({"Unnamed: 0":"a"}, axis=1)
bikecounts_df = bikecounts_df.drop(['a','Date','Day'],axis=1)

In [23]:
bikecounts_df.head()

High Temp (°F)  Low Temp (°F)  Precipitation  Brooklyn Bridge  \
0            78.1           66.0           0.01           1704.0   
1            55.0           48.9           0.15            827.0   
2            39.9           34.0           0.09            526.0   
3            44.1           33.1           0.47            521.0   
4            42.1           26.1           0.00           1416.0   

   Manhattan Bridge  Williamsburg Bridge  Queensboro Bridge  Total  
0              3126               4115.0             2552.0  11497  
1              1646               2565.0             1884.0   6922  
2              1232               1695.0             1306.0   4759  
3              1067               1440.0             1307.0   4335  
4              2617               3081.0             2357.0   9471

In [7]:
bikecounts_df['Brooklyn Bridge'].sum()

68089.0

In [8]:
bikecounts_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 8 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   High Temp (°F)       30 non-null     float64
 1   Low Temp (°F)        30 non-null     float64
 2   Precipitation        30 non-null     float64
 3   Brooklyn Bridge      30 non-null     float64
 4   Manhattan Bridge     30 non-null     int64  
 5   Williamsburg Bridge  30 non-null     float64
 6   Queensboro Bridge    30 non-null     float64
 7   Total                30 non-null     int64  
dtypes: float64(6), int64(2)
memory usage: 2.0 KB


In [9]:
bikecounts_df.describe()

High Temp (°F)  Low Temp (°F)  Precipitation  Brooklyn Bridge  \
count       30.000000      30.000000      30.000000        30.000000   
mean        60.580000      46.413333       0.052333      2269.633333   
std         11.347289       9.662502       0.103647       995.633241   
min         39.900000      26.100000       0.000000       504.000000   
25%         55.500000      44.100000       0.000000      1511.250000   
50%         62.100000      46.900000       0.000000      2379.500000   
75%         68.000000      50.000000       0.080000      3104.000000   
max         81.000000      66.000000       0.470000      3871.000000   

       Manhattan Bridge  Williamsburg Bridge  Queensboro Bridge         Total  
count         30.000000            30.000000          30.000000     30.000000  
mean        4049.533333          4862.466667        3352.866667  14534.500000  
std         1729.740976          1840.652747        1115.381263   5650.877227  
min          997.000000          1440.000000        1306.000000   4335.000000  
25%         2744.250000          3425.500000        2480.750000  10071.250000  
50%         4165.000000          5194.000000        3477.000000  15292.500000  
75%         5249.750000          6021.250000        4172.750000  18281.250000  
max         6951.000000          7834.000000        5032.000000  23318.000000

In [21]:
# Select your independent X terms, and your dependent y term. 
X = bikecounts_df[["High Temp (°F)","Low Temp (°F)","Precipitation"]]
y = bikecounts_df['Brooklyn Bridge']
X = sm.add_constant(X)
model = sm.OLS(y, X)
results = model.fit()

In [22]:
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Brooklyn Bridge   R-squared:                       0.712
Model:                            OLS   Adj. R-squared:                  0.679
Method:                 Least Squares   F-statistic:                     21.46
Date:                Fri, 26 Mar 2021   Prob (F-statistic):           3.31e-07
Time:                        20:13:13   Log-Likelihood:                -230.47
No. Observations:                  30   AIC:                             468.9
Df Residuals:                      26   BIC:                             474.5
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
const           -718.7735    626.377     -1.

# Explanation

The Null Hypothesis: 
1. The number of bikes crossing Brooklyn Bridge is dependent on Precipitation and Temperature.

The Alternative Hypothesis:
1. The number of bikes crossing Brooklyn Bridge is not dependent on Precipitation and Temperature.

#Analysis of the OLS Regression Results (P-value)
1. High Temp (°F): Pvalue = 0.000 indicates a high statistically significant correlation 
   between high temperature and bikes crossing Brooklyn Bridge.
   
3. Low Temp (°F): Pvalue = 0.104 is above the significance level of 0.05, and may not have much
   significance in determining the frequency at which bikes cross the Brooklyn Bridge.
   
2. Precipitation: Pvalue = 0.006 indicates a high statistically significant correlation
   between precipitation and bikes crossing Brooklyn Bridge.
   
A p-value less than 0.05 is statistically significant. It indicates strong evidence against the null hypothesis, as there is less than a 5% probability the null is correct and therefore the results are random. 

We therefore reject the null hypothesis, and accept the alternative hypothesis.
